### ISSUE TITLE
IWAE bound in VAE example is not correct?

### ISSUE THREAD

**Issue by amatsukawa** [tensorflow/probability#189]
I believe this line might be wrong?
https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/vae.py#L388

Unlike the ELBO bound, the negative IWAE bound has a sum, ie. -E[log 1/k SUM ...] so the negative in front of the expectation cannot be moved to flip the `p(x, z)/q(z|x)` ratio like the ELBO, which this line effectively does by reusing the `elbo_local` tensor.

**Comment by jvdillon** [comment]
Good pt. Can I ask that you make a PR to fix this? :)

**Comment by SiegeLordEx** [comment]
I think the code is correct as written. `elbo_local` is exactly `log (p(x, z)/q(z|x))` (e.g. follow the sign starting from the computation of `distortion`), so using it for IWAE bound computation is correct: we're not computing the negative IWAE bound. The negative sign in front of the `elbo_local` RHS expression is there just because of the definitions of `rate` and `distortion`, it's not there to negate the bound. The only place we actually negate the bound a few lines below when we do `loss = -elbo`.

**Comment by axch** [comment]
@SiegeLordEx Do you think this is worth adding a clarifying comment to the example, or do you want to just close this issue as-is?

**Comment by SiegeLordEx** [comment]
I think the code is more confusing than usual since it uses the rate + distortion formalism at the same time as elbo, resulting in quite a few extra negative signs. Rate + distortion formalism is a useful didactic tool though, so I think we're just in an unfortunate local optimum. I recommend keeping it as is, but if there's ideas, there's always an option to change things.
